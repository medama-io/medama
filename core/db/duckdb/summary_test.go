package duckdb_test

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/medama-io/medama/db"
)

func TestGetWebsiteSummary(t *testing.T) {
	_, require, ctx, client := UseDatabaseFixture(t, SimpleFixture)
	summary, err := client.GetWebsiteSummary(ctx, &db.Filters{
		Hostname:    MediumHostname,
		PeriodStart: TimeStart,
		PeriodEnd:   TimeEnd,
	})
	require.NoError(err)

	snaps.MatchJSON(t, summary)
}

func TestGetWebsiteSummaryEmpty(t *testing.T) {
	assert, require, ctx, client := UseDatabaseFixture(t, SimpleFixture)

	summary, err := client.GetWebsiteSummary(ctx, &db.Filters{
		Hostname:    DoesNotExistHostname,
		PeriodStart: TimeStart,
		PeriodEnd:   TimeEnd,
	})
	require.NoError(err)

	assert.Equal(0, summary.Visitors)
	assert.Equal(0, summary.Pageviews)
	assert.Equal(float32(0), summary.BounceRate)
	assert.Equal(0, summary.Duration)
}

func TestGetWebsiteSummaryFilterAll(t *testing.T) {
	_, require, ctx, client := UseDatabaseFixture(t, SimpleFixture)

	for _, filter := range generateFilterAll(MediumHostname) {
		summary, err := client.GetWebsiteSummary(ctx, filter.Filters)
		require.NoError(err)

		snaps.MatchJSON(t, summary)
	}
}

// TODO: Need to make a new deterministic time fixture for these tests.
// func TestGetWebsiteIntervals(t *testing.T) {}
// func TestGetWebsiteLast24Hours(t *testing.T) {}
