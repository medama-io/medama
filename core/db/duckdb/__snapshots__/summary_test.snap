
[TestGetWebsiteSummary - 1]
{
 "Bounces": 124684,
 "Duration": 5003,
 "Pageviews": 1000000,
 "Visitors": 499657
}
---

[TestGetWebsiteSummaryFilterAll - 1]
{
 "Bounces": 41640,
 "Duration": 4980,
 "Pageviews": 332879,
 "Visitors": 166090
}
---

[TestGetWebsiteSummaryFilterAll - 2]
{
 "Bounces": 13844,
 "Duration": 4992,
 "Pageviews": 110733,
 "Visitors": 55243
}
---

[TestGetWebsiteSummaryFilterAll - 3]
{
 "Bounces": 4691,
 "Duration": 4997,
 "Pageviews": 37170,
 "Visitors": 18661
}
---

[TestGetWebsiteSummaryFilterAll - 4]
{
 "Bounces": 2355,
 "Duration": 5003,
 "Pageviews": 18789,
 "Visitors": 9486
}
---

[TestGetWebsiteSummaryFilterAll - 5]
{
 "Bounces": 783,
 "Duration": 4971,
 "Pageviews": 6165,
 "Visitors": 3116
}
---

[TestGetWebsiteSummaryFilterAll - 6]
{
 "Bounces": 268,
 "Duration": 5098,
 "Pageviews": 2113,
 "Visitors": 1090
}
---

[TestGetWebsiteSummaryFilterAll - 7]
{
 "Bounces": 96,
 "Duration": 4728,
 "Pageviews": 726,
 "Visitors": 381
}
---

[TestGetWebsiteSummaryFilterAll - 8]
{
 "Bounces": 31,
 "Duration": 4732,
 "Pageviews": 220,
 "Visitors": 116
}
---

[TestGetWebsiteSummaryFilterAll - 9]
{
 "Bounces": 6,
 "Duration": 6547,
 "Pageviews": 67,
 "Visitors": 35
}
---

[TestGetWebsiteSummaryFilterAll - 10]
{
 "Bounces": 3,
 "Duration": 3389,
 "Pageviews": 28,
 "Visitors": 17
}
---
